c10/util/Optional.h

```diff
@@ -587,13 +587,16 @@ class optional : private OptionalBase<T> {
   constexpr bool initialized() const noexcept {
     return OptionalBase<T>::initialized();
   }
+
+ public:
```

Comment:
I had to make `dataptr()` public now that `OptionalScalarRef` stores `const Scalar*`. But it doesn't seem right to do this.